samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/facade/AssessmentGradingFacadeQueries.java

```diff
@@ -513,32 +513,36 @@ public Long saveMedia(byte[] media, String mimeType) {
         return saveMedia(mediaData);
     }
 
-    protected void pushAdvisor() {
-        securityService.pushAdvisor(new SecurityAdvisor() {
+    protected SecurityAdvisor pushAdvisor() {
+        SecurityAdvisor samigoAdvisor = new SecurityAdvisor() {
             public SecurityAdvice isAllowed(String userId, String function, String reference) {
                 return SecurityAdvice.ALLOWED;
             }
-        });
+        };
+        securityService.pushAdvisor(samigoAdvisor);
+        return samigoAdvisor;
     }
 
-    protected void popAdvisor() {
-        securityService.popAdvisor();
+    protected void popAdvisor(SecurityAdvisor sa) {
+        if (sa != null) {
+            securityService.popAdvisor(sa);
+        }
```

Comment:
I wonder if silently continuing is the best option here? Just wondering when someone does another refactor later on and gets something a little wrong. Maybe throw `IllegalArgumentException`?

Reply:
Cool, I will change to throw if sa == null